src/core/test/system/lender/redeem.t.sol

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2019 Centrifuge
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.12;
+
+import "../system.t.sol";
+import {SwitchableDistributor} from "../../../lender/distributor/switchable.sol";
+
+contract RedeemTest is SystemTest {
```

Comment:
```suggestion
contract RedeemTest is TestUtils {
```

If we use system tests we execute all system test twice.

But maybe it is a good pattern to move a default `setUp` method into TestUtils

Reply:
after conversation with @ilinzweilin and @manuelpolzhofer we have decided to move this issue to another PR

Reply:
yes, I would even create a clean base SystemTest contract that inherits from testUtils but also has some additional methods like setUp, assertPrecondition, assertPostcondition... and let the tests inherit from it.